### DISCUSSION TITLE
FPS drop on kirkstone l4tr32.7.

### DISCUSSION THREAD

**Discussion by yrodriguez31** [OE4T/meta-tegra#1739] [Q&A]
Hello, 
I'm working on the branch kirkstone-l4t-r32.7, which includes CUDA 10.2.
Currently, we're experiencing a performance issue with the Yocto image, as we see a 5 fps drop compared to the L4T Jetpack R32.7.4 in GPU processing. I've tried every optimization I know and cleaned up the image as much as possible. Using jtop, I observe that the GPU hardware configuration is identical, but I still notice that in Yocto, we are using more memory and CPU but less GPU.

In my Yocto configuration, I’ve made sure to require GStreamer 1.14 for hardware encoders, and I’ve set CUDA_GCCVERSION = "8.%" so that nvcc uses GCC 8. However, I still can’t regain the 5 fps. In CMake, I used the following optimizations:
`
CUDA_PATH = "/usr/local/cuda-10.2"
EXTRA_OECMAKE += "-DCMAKE_BUILD_TYPE=Release \
                 -DCMAKE_C_FLAGS_RELEASE='-O3 -march=armv8-a -mtune=cortex-a72 -flto -DNDEBUG' \
                 -DCMAKE_CXX_FLAGS_RELEASE='-O3 -march=armv8-a -mtune=cortex-a72 -flto -DNDEBUG' \
                 -DPROJECT_VERSION=${PV} \
                 -DCUDA_TOOLKIT_ROOT_DIR=${CUDA_PATH} \
                 -DCUDA_INCLUDE_DIRS=${CUDA_PATH}/include \
                 -DCUDA_CUDART_LIBRARY=${CUDA_PATH}/lib64/libcudart.so"
`

Any idea where this issue might come from? Have you ever encountered this type of problem?

**Comment by yrodriguez31** [comment]
Hello ,
When i change the gpu conf with nvpmodel i have this message in dmesg:
`nvgpu: 17000000.gv11b                 tpc_pg_mask_store:843  [INFO]  no value change, same mask already set`

And when i compare my os yocto and my os with the jetpack it's field like the os on yocto is stuck at 10W

Regards

**Reply by madisongh** [reply]
Does the `nvpmodel` service start up without errors at boot time?  IIRC, even in the R32-series BSP, the GPU driver really wants to have the power model applied before anything else touches the GPU, but after the driver is loaded, so if you've made changes to the boot sequence that might have affected that order, that could be the problem.

**Reply by yrodriguez31** [reply]
Hello , yes the nvpmodel start without errors . I don't think that a touch the boot order but i can check with kernel logs.
Thanks

**Reply by madisongh** [reply]
OK.  Also, make sure that when you're comparing your build against stock L4T, that you've configured the same model in both cases, and if stock L4T runs their `jetson_clocks` script by default (I don't remember now if they do), that you do the same.
